File: packages/expo-eas-client-id/ios/EASClientId/EASClientIdModule.swift
// Copyright 2015-present 650 Industries. All rights reserved.

import ExpoModulesCore

public class EASClientIdModule: Module {
  public func definition() -> ModuleDefinition {
    name("EASClientId")

    function("getClientIdAsync") {
      EASClientId.uuid.uuidString
    }
  }
}

File: packages/expo-eas-client-id/android/src/main/java/expo/modules/easclientid/EASClientIdModule.kt
package expo.modules.easclientid

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class EASClientIdModule : Module() {
  private val context
    get() = requireNotNull(appContext.reactContext) {
      "React Application Context is null"
    }

  override fun definition() = ModuleDefinition {
    name("EASClientId")

    function("getClientIdAsync") {
      EASClientId(context).uuid.toString()
    }
  }
}
